packages/blocks/src/_common/components/tooltip/tooltip.ts

```diff
@@ -161,74 +162,82 @@ export class Tooltip extends LitElement {
   @property({ attribute: false })
   accessor allowInteractive = false;
 
-  private _hoverController = new HoverController(
-    this,
-    () => {
-      // const parentElement = this.parentElement;
-      // if (
-      //   parentElement &&
-      //   'disabled' in parentElement &&
-      //   parentElement.disabled
-      // )
-      //   return null;
-      if (this.hidden) return null;
-      let arrowStyles: StyleInfo = {};
-      return {
-        template: ({ positionSlot, updatePortal }) => {
-          positionSlot.on(data => {
-            // The tooltip placement may change,
-            // so we need to update the arrow position
-            if (this.arrow) {
-              arrowStyles = updateArrowStyles(data);
-            } else {
-              arrowStyles = {};
-            }
-            updatePortal();
-          });
+  @property({ attribute: false })
+  accessor hoverOptions: Partial<HoverOptions> = {};
 
-          const slot = this.shadowRoot?.querySelector('slot');
-          if (!slot) throw new Error('slot not found in tooltip!');
-          // slot.addEventListener('slotchange', () => updatePortal, {
-          //   once: true,
-          // });
-          const slottedChildren = slot
-            .assignedNodes()
-            .map(node => node.cloneNode(true));
-          return html`
-            <style>
-              ${this._getStyles()}
-            </style>
-            <div class="affine-tooltip" role="tooltip">${slottedChildren}</div>
-            <div class="arrow" style=${styleMap(arrowStyles)}></div>
-          `;
-        },
-        computePosition: portalRoot => ({
-          referenceElement: this.parentElement!,
-          placement: this.placement,
-          middleware: [
-            this.autoFlip && flip({ padding: 12 }),
-            offset((this.arrow ? TRIANGLE_HEIGHT : 0) + this.offset),
-            arrow({
-              element: portalRoot.shadowRoot!.querySelector('.arrow')!,
-            }),
-          ],
-          autoUpdate: true,
-        }),
-      };
-    },
-    {
-      leaveDelay: 0,
-      // The tooltip is not interactive by default
-      safeBridge: false,
-      allowMultiple: true,
-    }
-  );
+  private _hoverController!: HoverController;
 
   override connectedCallback() {
     super.connectedCallback();
     const parent = this.parentElement;
     assertExists(parent, 'Tooltip must have a parent element');
 
+    this._hoverController = new HoverController(
```

Comment:
Maybe move to a function called `setupHoverController` to keep the connectedCallback clean?